wile/cert.py

```diff
@@ -184,18 +202,47 @@ def _get_http_challenge(ctx, authzr):
     ctx.fail('no acceptable challenge type found; only HTTP01 supported')
 
 
-def _store_webroot_validation(webroot, challb, val):
+def _store_webroot_validation(ctx, remote, webroot, challb, val):
     logger.info('storing validation of %s', webroot)
-    try:
-        os.makedirs(os.path.join(webroot, challb.URI_ROOT_PATH), 0o755)
-    except OSError as e:
-        if e.errno != errno.EEXIST:
-            raise
-
-    with open(os.path.join(webroot, challb.path.strip('/')), 'wb') as outf:
-        logger.info('storing validation to %s', outf.name)
-        outf.write(b(val))
-        atexit.register(os.unlink, outf.name)
+    chall_path = os.path.join(webroot, challb.path.strip('/'))
+    if not remote:
+        try:
+            os.makedirs(os.path.join(webroot, challb.URI_ROOT_PATH), 0o755)
+        except OSError as e:
+            if e.errno != errno.EEXIST:
+                raise
+
+        with open(chall_path, 'wb') as outf:
+            logger.info('storing validation to %s', outf.name)
+            outf.write(b(val))
+            atexit.register(os.unlink, outf.name)
+    else:
+        username = remote[0] is not None and remote[0] or os.environ.get('USER')
+        hostname = remote[1]
+        port = remote[2] is not None and int(remote[2]) or 22
+        private_key_path = os.path.expanduser(remote[3])
+        private_key_pass = remote[4] is not '' and remote[4] or None
+        private_key_type = remote[5].upper()
```

Comment:
These lines are also a bit hard on the eyes with so many index accesses. Re-reading this in a few days might prove difficult 😉
This problem would "automatically" disappear with the approach mentioned above at `_DomainWebrootType`.